function/random/random.js

```diff
@@ -3,21 +3,60 @@ module.exports = function(RED) {
     "use strict";
     function RandomNode(n) {
         RED.nodes.createNode(this,n);
-        this.low = Number(n.low || 1);
-        this.high = Number(n.high || 10);
+
+        this.low = n.low
+        this.high = n.high
         this.inte = n.inte || false;
         this.property = n.property||"payload";
         var node = this;
+        var tmp = {};
+
         this.on("input", function(msg) {
-            var value;
-            if (node.inte == "true" || node.inte === true) {
-                value = Math.round(Math.random() * (node.high - node.low + 1) + node.low - 0.5);
-            }
-            else {
-                value = Math.random() * (node.high - node.low) + node.low;
-            }
-            RED.util.setMessageProperty(msg,node.property,value);
-            node.send(msg);
+
+           if (node.low) {            // if the the node has a value use it
+               tmp.low = node.low
+           } else {                   // if 'from' in the msg, use it or default to 1
+               tmp.low = ('from' in msg) ? msg.from : 1;
+           }
+
+           if (node.high) {           // if the the node has a value use it
+               tmp.high = node.high
+           } else {                   // if 'to' in the msg, use it or default to 1
+               tmp.high = ('to' in msg) ? msg.to : 10;
+           }
+           
+          // if tmp.low or high are not numbers, send an error msg
+          if ( (isNaN(tmp.low)) || (isNaN(tmp.high)) ) {
+              this.error({node:"random",text:"one of the input values is not a number"})
+          } 
+          else {
+             // force tmp.high/low to Numbers since they may be a string
+             tmp.low = Number(tmp.low)
```

Comment:
Does it really matter if if it is done before the test to see if they are numbers? 
         // if tmp.low or high are not numbers, send an error msg
           if ( (isNaN(tmp.low)) || (isNaN(tmp.high)) ) {
               this.error({node:"random",text:"one of the input values is not a number"})
           } 
           else {
              // force tmp.high/low to Numbers since they may be a string
              tmp.low = Number(tmp.low)

Reply:
probably not but this is javascript so there may be some weird edge cases where isNaN says is an OK number then Number fails to convert it.